Core/automation/lib/javascript/core/log.js

```diff
@@ -0,0 +1,208 @@
+            _name = _name.slice(-3) == ".js" ? _name.slice(0,-3) : null;
+        }
+        var _logger  = Java.type("org.slf4j.LoggerFactory").getLogger(_name === null ? LOGGER_PREFIX : LOGGER_PREFIX + "." + _name.toString().toLowerCase());
+
+        try {
+            // Set default config for config params not provided.
+            if (_config === undefined) _config = {};
+            if (_config.ERROR === undefined) _config.ERROR = {};
+            if (_config.WARN  === undefined) _config.WARN  = {};
+            if (_config.INFO  === undefined) _config.INFO  = {};
+            if (_config.DEBUG === undefined) _config.DEBUG = {};
+            if (_config.ERROR.prefix === undefined) _config.ERROR.prefix = "short";
+            if (_config.WARN.prefix  === undefined) _config.WARN.prefix  = "none";
+            if (_config.INFO.prefix  === undefined) _config.INFO.prefix  = "none";
+            if (_config.DEBUG.prefix === undefined) _config.DEBUG.prefix = "short";
+
+            var _MessageFormatter = Java.type("org.slf4j.helpers.MessageFormatter");
+
+            var _getLogMessage = function _getLogMessage (msg, prefix, levelString) {
+                if (msg instanceof Error) {
+                    msg.caller = msg.stack.split('\n\tat ')[1].split(' (')[0]
+                } else {
+                    msg = Error(msg);
+                    msg = _getCallerDetails(msg);
+                }
+                msg = _legacyLoggerCorrection(msg);
+
+                if (prefix === undefined) prefix = "log";
+
+                if (prefix == "none") {
+                    return msg.message;
+                }
+
+                var level = "";
+                var nameText = "";
+                if (prefix != "log") {
+                    level = "[" + levelString + "] ";
+                    nameText = _name === null ? "" : _name;
+                }
+
+                if (prefix == "level") {
+                    return (level + msg.message);
+                }
+
+                var callerText;
+                if (msg.caller.substr(0,1) == "<") {
+                    callerText = ", in " + msg.caller;
+                } else {
+                    callerText = ", function " + msg.caller;
+                }
+
+                var message = msg.message == "" ? "" : msg.message + " ";
+
+                if (prefix == "short") {
+                    return (level + message + "\t\t[" + (nameText != "" ? nameText + ", " : "") + msg.fileName.split('/').pop() + ":" + msg.lineNumber + callerText + "]");
+                }
+                return (level + message + "\t\t[" + (nameText != "" ? nameText + ": " : "") + "at source " + msg.fileName.split(AUTOMATION_PATH).pop() + ", line " + msg.lineNumber + callerText + "]");
+            }
+
+            var _getCallerDetails = function _getCallerDetails (msg) {
+                var matches = msg.stack.split('\n\tat ')[3].match(/(.+?) \((.+):(\d+?)\)/);
```

Comment:
Same comment as below - this stack parsing is specific to Nashorn; `/\s+at\s+/` should work for Nashorn + others.

Reply:
Also, same comment as below.